project_name/models/grid_search_model.py

```diff
@@ -0,0 +1,98 @@
+import pandas as pd
+import itertools
+from features.embedding_processing import balance_dataframe, reduce_dimensions
+from models.nn_classifier import create_model, create_optimizer, train_model
+
+
+class GridSearchModel():
+    """
+    A model that performs a grid search over a set of hyperparameters using a neural network
+    """
+
+    hyperparameters = {
+        "n_dimensions": [10, 50],
+        "n_layers": [1, 2],
+        "n_neurons": [10, 50],
+        "optimizer_type": ["adam", "sgd"],
+        "learning_rate": [0.01, 0.001],
+        "epochs": [5, 10, 20],
+    }
+
+    def __init__(self):
+        """
+        Initialize the model, load data and create results dataframe
+        """
+
+        # Load data, change to real embeddings later
+        self.train_data = pd.read_csv("../data/gold/train_random_embeddings.csv")
```

Comment:
maybe it would be nice to have the `_random_embeddings` in `train_random_embeddings` be a suffix parameter. in this class. if we later would like to switch from word2vec embeddings to bert or mobileBert, so it's easy – just change the `path_suffix` or `path_type` parameter.